streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -127,6 +133,11 @@ public void restore() {
                                       consumer.position(restorer.partition()),
                                       endOffsets.get(restorer.partition()));
                 }
+                // TODO: each consumer.position() call after seekToBeginning will cause
```

Comment:
Should probably raise a JIRA for this otherwise we will end up losing the performance gains that we got from batching everything else.